pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +379,54 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+# Det Diag Rewrite
+@register_canonicalize
+@register_stabilize
+@node_rewriter([det])
+def det_diag_rewrite(fgraph, node):
+    # Find if we have Blockwise Op
+    if not (isinstance(node.op, Blockwise) and isinstance(node.op.core_op, Det)):
+        return None
+
+    node_input_op = node.inputs[0]
+    # Check if the op is Elemwise and mul
+    if not (
+        isinstance(node_input_op.owner.op, Elemwise)
+        and isinstance(node_input_op.owner.op.scalar_op, Mul)
+    ):
```

Comment:
If you have a graph that is just `det(x)`, this will not have any owners and this will error. You need to add a check for this:

```
if not (
    node_input_op.owner is not None
    isinstance(node_input_op.owner.op, Elemwise)
    and isinstance(node_input_op.owner.op.scalar_op, Mul)
):